src/core/Akka.Remote/RemoteActorRefProvider.cs

```diff
@@ -403,13 +403,25 @@ internal IInternalActorRef ResolveActorRefWithLocalAddress(string path, Address
                         return RootGuardian;
                     return _local.ResolveActorRef(RootGuardian, actorPath.ElementsWithUid);
                 }
-                    
-                return new RemoteActorRef(Transport, localAddress, new RootActorPath(actorPath.Address) / actorPath.ElementsWithUid, ActorRefs.Nobody, Props.None, Deploy.None);
+
+                return CreateRemoteRef(actorPath, localAddress);
             }
             _log.Debug("resolve of unknown path [{0}] failed", path);
             return InternalDeadLetters;
         }
 
+        
+        /// <summary>
+        /// Used to create <see cref="RemoteActorRef"/> instances upon deserialiation inside the Akka.Remote pipeline.
+        /// </summary>
+        /// <param name="actorPath">The remote path of the actor on its physical location on the network.</param>
+        /// <param name="localAddress">The local path of the actor.</param>
+        /// <returns>An <see cref="IInternalActorRef"/> instance.</returns>
+        protected virtual IInternalActorRef CreateRemoteRef(ActorPath actorPath, Address localAddress)
```

Comment:
Needed to expose a factory method here inside the `RemoteActorRefProvider` that for changing the underlying `RemoteActorRef` implementation to a version that can be instrumented. This is stuff we're using for remote propagation of tracing information mostly.